tools/cli/pkg/ers/command/instances.go

```diff
@@ -42,17 +45,37 @@ type Filters struct {
 
 type InstancesCommand struct {
 	corbaCmd        *cobra.Command
-	instanceFetcher InstanceFetcher
+	instanceFetcher fetcher.InstanceFetcher
 	filters         Filters
 	source          string
+	pageStart       int
+	pageSize        int
+	pageLimit       int
 }
 
 func (c *InstancesCommand) Run() error {
+	if GlobalOpts.ClientID() == "" ||
+		GlobalOpts.ClientSecret() == "" ||
+		GlobalOpts.OauthUrl() == "" {
+		return errors.New("no auth data provided")
+	}
+
+	config := clientcredentials.Config{
+		ClientID:     GlobalOpts.ClientID(),
+		ClientSecret: GlobalOpts.ClientSecret(),
+		TokenURL:     GlobalOpts.OauthUrl(),
+	}
+	configClient := config.Client(context.Background())
+
 	if c.source != "" {
-		c.instanceFetcher = client.NewFileClient(c.source)
+		c.instanceFetcher = fetcher.NewFileClient(c.source)
 	} else {
+		// create a shared ERS HTTP client which does the oauth flow
+		httpClient := client.NewHTTPClient(logger.New(), configClient)
```

Comment:
as I see, the http client is not shared. for instances operation it does not matter, but for migration - yes